specs/DesignDiscussions/FocusLostValidation.md

```diff
@@ -0,0 +1,132 @@
+
+In version 1.3 of the Adaptive Cards schema we introduced input validation in Adaptive Cards. For that initial version we validated inputs only when the user presses the submit button. In order to improve the user experience, and to allow more control to the hosts, this spec covers the introduction of additional options for validation.
+
+
+## Table of Contents
+- [Schema Changes](#schema-changes)
+- [Behavior](#Behavior)
+    - [First Validation](#First-validation)
+        -  [onFocusLost](#onFocusLost)
+        - [onFocusLostWithInput](#onFocusLostWithInput)
+        - [onSubmit](#onSubmit)
+    - [Subsequent Validations](#Subsequent-Validations)
+- [Motivation](#Motivation)
+    - [Research](#Research)
+    - [WinUI Consistency](#WinUI-Consistency)
+    - [Survey Data](#Survey-Data)
+
+## Schema Changes
+
+This feature does not introduce any changes to the Adaptive Cards schema, but does add properties to the Host Config schema. Specifically, the `validationBehavior` property is added to `InputsConfig` as described below:
+
+```json
+{
+    "InputsConfig": {
+        "properties": {
+            "validationBehavior":{
+                "type":"string",
+                "description":"Controls when inputs are validated",
+                "default":"onFocusLost",
+                "enum":[
+                    "onFocusLost",
+                    "onFocusLostWithInput",
+                    "onSubmit"
+                ]
+            }
+        }
+    }
+},
+```
+
+These values allow hosts to determine the behavior of input validation as described below.
+
+## Behavior
+
+### First validation
+The host config values above determine when the first validation of the property takes place.
+
+#### onFocusLost
+The input is validated when it loses focus for the first time. If a user tabs past a required input, or enters an invalid value in field, the error message will show once focus leaves that field.
+
+#### onFocusLostWithInput
+The input is validated when it loses focus only if the user has entered a value into the input field. If this value is set we will not validate for the `isRequired` property on focus lost unless the user has typed something and subsequently removed it. 
+
+#### onSubmit
+This is the behavior that shipped with the 1.3 schema update. The inputs are validated when the submit button is pressed. No validation takes place before submit is pressed.
+
+### Subsequent Validations
+Once the field has been validated for the first time based on the behavior specified by the host config, all host config settings have the same subsequent behavior. Once the field has been marked as invalid, from that point forward it will be validated on each keystroke. This allows users to correct their errors and see when they have successfully met the validation criteria. Validation will continue to be run on each keystroke even after the error has been corrected, in order to allow users to continue to see if their new value is correct.
```

Comment:
is this a per field behaviour or a general validation? Take for example fields A and field B (maybe answer for B depends on answer for A), if only validation for A fails and the value has to be changed for B (now B will be invalid) it may seem odd that field A validates on each keystroke and B doesn't